spec/spec.md

```diff
@@ -209,6 +209,27 @@ The type of the `If` expression is the common type between `A` and `B`.
 **Expression kind**: r-value
 **Uses**: `cond`, `body`, and - if present - `else`
 
+#### `While`
+
+```grammar
+expr += (While cond:<expr> body:<expr>)
+```
+
+Repeatedly evaluates `body`, as long as `cond` evaluates to `true`. Both
+`body` and `cond` are part of a new scope.
+
+Let `C` be the type of `cond` and let `T` be the type of `body`. An error is
+reported if:
+* `C` is not `bool`
+* `T` is not `unit`
```

Comment:
presuming that a hypothetical `quit` procedure would be `() -> void`, then wouldn't this definition be problematic, as one couldn't put `quit` as the `body`?

N.B.: I'm ok punting on this and figuring it out later

Reply:
Hm, you're right. I'm not sure what I was thinking when disallowing `void` here. Not only a hypothetical `quit` cannot be put as the `body`, a `Return` or `Unreachable` cannot be put there too.

I'll adjust the rules.